backend/src/main/java/com/htecgroup/skynest/service/impl/FileServiceImpl.java

```diff
@@ -66,6 +70,31 @@ public FileResponse uploadFile(MultipartFile multipartFile, UUID bucketId) {
     checkBucketNotDeleted(emptyFileMetadata);
     checkBucketSizeExceedsMax(emptyFileMetadata);
 
+    return uploadFile(emptyFileMetadata, multipartFile);
+  }
+
+  @Override
+  @Transactional(rollbackFor = Exception.class)
+  public FileResponse uploadFileToFolder(MultipartFile multipartFile, UUID folderId) {
+
+    FileMetadataEntity emptyFileMetadata =
+        initFileMetadataWithFolder(
+            multipartFile.getOriginalFilename(),
+            multipartFile.getSize(),
+            multipartFile.getContentType(),
+            folderId);
+
+    checkOnlyCreatorsCanAccessPrivateBuckets(emptyFileMetadata);
```

Comment:
Just a note:

I would consider when refactoring this class to explore the option for something like emptyFileMetadata.validate(). Or maybe to extract this in some class FileMetadataDto or however you are naming the models for shifting data between repository and controller layers and make a validate method there